src/main/scala/mesosphere/marathon/core/storage/store/impl/zk/RichCuratorFramework.scala

```diff
@@ -159,22 +158,25 @@ class RichCuratorFramework(val client: CuratorFramework) {
     */
   @SuppressWarnings(Array("CatchFatal"))
   def blockUntilConnected(lifecycleState: LifecycleState): Unit = {
-    val timeoutAt: Long = System.currentTimeMillis() + client.getZookeeperClient.getConnectionTimeoutMs
+    val zkTimeout: Int = client.getZookeeperClient.getConnectionTimeoutMs
+    val timeoutAt: Long = System.currentTimeMillis() + zkTimeout
+    var connected = false
 
-    @tailrec def poll(): Unit = {
-      if (System.currentTimeMillis > timeoutAt)
-        throw new InterruptedException("timed out while waiting for zookeeper connection")
-      else if (!lifecycleState.isRunning)
+    while (!connected && System.currentTimeMillis <= timeoutAt) {
```

Comment:
Do you think we should use some kind of exponential backoff here to limit amount of requests we send?

Reply:
I don't think it will add any real value here, because it is supposed to happen only once during start-up, and it is supposed to fail after the predefined timeout.